File: lib/domain/public_plan/exceptions.rb
# frozen_string_literal: true

module Domain
  module PublicPlan
    # 公開エントリスケジュール API 共通（レスキュー経路との互換で Error サフィックスを維持）
    module Exceptions
      WeatherLocationMissingError = Class.new(StandardError)
      PredictionPayloadMissingError = Class.new(StandardError)
      WeatherPredictionFailedError = Class.new(StandardError)
    end
  end
end

File: crates/agrr-domain/src/public_plan/exceptions.rs
//! Ruby: `Domain::PublicPlan::Exceptions`

use thiserror::Error;

/// Ruby: `WeatherLocationMissingError`
#[derive(Debug, Error, Clone, PartialEq, Eq)]
#[error("weather location missing")]
pub struct WeatherLocationMissingError;

/// Ruby: `PredictionPayloadMissingError`
#[derive(Debug, Error, Clone, PartialEq, Eq)]
#[error("prediction payload missing")]
pub struct PredictionPayloadMissingError;

/// Ruby: `WeatherPredictionFailedError`
#[derive(Debug, Error, Clone, PartialEq, Eq)]
#[error("{0}")]
pub struct WeatherPredictionFailedError(pub String);
